hack/shoot-comparator/pkg/shoot/matcher_test.go

```diff
@@ -107,5 +134,986 @@ var _ = Describe(":: shoot matcher :: ", func() {
 			})),
 			false,
 		),
+		Entry(
+			"should detect differences in spec #2",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Region: "test1",
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				DNS:    &v1beta1.DNS{},
+				Region: "test1",
+			})),
+			false,
+		),
+		Entry(
+			"should detect differences in spec/exposureClassName #1",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				ExposureClassName: ptr.To[string]("mage"),
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				ExposureClassName: ptr.To[string]("bard"),
+			})),
+			false,
+		),
+		Entry(
+			"should detect no differences in spec/exposureClassName #1",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				ExposureClassName: ptr.To[string]("mage"),
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				ExposureClassName: ptr.To[string]("mage"),
+			})),
+			true,
+		),
+		Entry(
+			"should detect differences in spec/dns #1",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
```

Comment:
`dns.providers.domains.include` is not covered. The matcher compares it so we should include it in the test cases.

Reply:
`dns.providers.domains.exclude` and `dns.providers.zones` are not covered. The matcher ignore that field so we need to make sure it won't be taken to the comparison.

Reply:
We need to also cover the case with `nil` matcher returned from `newDNSMatcher` function.